### PR TITLE
[CoreBundle] Move serialization to AdminApiBundle and move ProductVar…

### LINKED ISSUE #8087: [Product][Serializer] ProductVariant and Taxon serializers has to be not dependent on SyliusAdminApiBundle
In fact, i disabled Api in my Sylius installation because i do not need it for the moment.

In product admin form, in associations, autocomplete use serializer which defines relations in that way : 

```yaml 
 relations:
        -   rel: self
            href:
                route: sylius_admin_api_product_variant_show
                parameters:
                    code: expr(object.getCode())
                    productCode: expr(object.getProduct().getCode())
                    version: 1
        -   rel: product
            href:
                route: sylius_admin_api_product_show
                parameters:
                    code: expr(object.getProduct().getCode())
                    version: 1
            exclusion:
                groups: [Detailed]
```

I think routes like that has to depend on sylius_admin_ajax_*, WDYT ?

### PR DESCRIPTION
**Pull request by Niiko**
…iant relations serialization from ProductBundle to AdminApiBundle

| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | fixes #8087  |
| License         | MIT |

Move serialization from CoreBundle to AdminApiBundle
Move relations of ProductVariant serialization from ProductBundle to AdminApiBundle

I'm not sure to do it the right way @lchrusciel.

If i'm correct, every relations with admin_api routes have to be moved the same way no ?

### CODE REVIEW

**Comment by lchrusciel** [comment]
I don't better idea how to tackle it. Can you move relation from the rest of routes? Currently, the aren't useful anyway.

**Comment by lchrusciel** [comment]
We will need a note in UPGRADE file about it. And also, I'm pretty sure it is BC break, but we need it.

**Comment by Niiko** [comment]
Ok, i move a lot of relations, but on some, i have questions

- Country
- Province
- Zone
- Currency
- ExchangeRate
- Locale
- ProductAssociationType
- ProductAttribute
- ProductOption
- ShipmentUnit
- ShippingCategory
- TaxCategory

There are not in corebundle, do you want i create them ?

File: src/Sylius/Bundle/AdminApiBundle/Resources/config/serializer/Model.OrderItem.yml [hunk 1]
```diff
@@ -24,4 +32,4 @@ Sylius\Component\Core\Model\OrderItem:
             parameters:
                 code: expr(object.getVariant().getCode())
                 productCode: expr(object.getVariant().getProduct().getCode())
-                version: 1
+                version: 1
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminApiBundle/Resources/config/serializer/Model.OrderItem.yml]: Missing blank line at the end of file

File: src/Sylius/Bundle/AdminApiBundle/Resources/config/serializer/Model.OrderItem.yml [hunk 2]
```diff
@@ -6,6 +6,14 @@ Sylius\Component\Core\Model\OrderItem:
             expose: true
             groups: [Default, Detailed, DetailedCart]
     relations:
+        - rel: order
+          exclusion:
+            groups: [Default, Detailed, DetailedCart]
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminApiBundle/Resources/config/serializer/Model.OrderItem.yml]: Something went wrong with the indentation

File: src/Sylius/Bundle/AdminApiBundle/Resources/config/serializer/Model.ProductVariant.yml [hunk 3]
```diff
@@ -44,3 +44,19 @@ Sylius\Component\Core\Model\ProductVariant:
...
+                    productCode: expr(object.getProduct().getCode())
+                    version: 1
+        -   rel: product
+            href:
+                route: sylius_admin_api_product_show
+                parameters:
+                    code: expr(object.getProduct().getCode())
+                    version: 1
+            exclusion:
+                groups: [Detailed]
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminApiBundle/Resources/config/serializer/Model.ProductVariant.yml]: Missing blank line at the end of file

File: UPGRADE.md [hunk 4]
```diff
@@ -156,20 +156,79 @@
   * from `SyliusCoreBundle:Email:userRegistration.html.twig` to `SyliusShopBundle:Email:userRegistration.html.twig`
   * from `SyliusCoreBundle:Email:passwordReset.html.twig` to `SyliusShopBundle:Email:passwordReset.html.twig`
   * from `SyliusCoreBundle:Email:verification.html.twig` to `SyliusShopBundle:Email:verification.html.twig`
-  
+
+* The following serialization configuration was moved from CoreBundle/Resources/config/app/config.yml to AdminApiBundle/Resources/config/app/config.yml
```
Comment (lchrusciel) [inline UPGRADE.md]: It should b part of migration from `FROM 1.0.0-beta.2 to 1.0.0` (new section)

Comment (Niiko) [inline UPGRADE.md]: my bad, read it too quickly

File: src/Sylius/Bundle/AdminApiBundle/Resources/config/serializer/Model.ProductVariant.yml (hunk 3 above)
Comment (Niiko) [inline src/Sylius/Bundle/AdminApiBundle/Resources/config/serializer/Model.ProductVariant.yml]: added

**Comment by pjedrzejewski** [comment]
Thank you Nicolas & Łukasz!

**Pull request merged**